internal/metrics/system.go

```diff
@@ -69,6 +69,12 @@ func (sm *SystemMetrics) Update(config models.DeviceConfigurationMessage) error
 	return nil
 }
 
+func (sm *SystemMetrics) Deregister() error {
```

Comment:
That method is already in `main` (https://github.com/jakub-dzon/k4e-device-worker/blob/7ff785b3ed54ec8f53173b982cc3851ad0469ae8/internal/metrics/system.go#L73). 
Please make sure that you work on the latest changes.